code/headers/display_module.hpp

```diff
@@ -1,25 +1,34 @@
 #pragma once
+// DO NOT RUN CLANG FORMAT ON THIS FILE 
 
 #include <base_module.hpp>
 #include <display_adapter.hpp>
 #include <hwlib.hpp>
 
 namespace r2d2::display {
+    template <std::size_t CursorCount, class DisplayScreen>
     class module_c : public base_module_c {
     protected:
-        display_c &display;
+        display_c<CursorCount, DisplayScreen> &display;
 
     public:
         /**
          * @param comm
          * @param led
          */
-        module_c(base_comm_c &comm, display_c &display)
+        module_c(base_comm_c &comm,
+                 display_c<CursorCount, DisplayScreen> &display)
             : base_module_c(comm), display(display) {
 
             // Set up listeners
             comm.listen_for_frames(
-                {r2d2::frame_type::DISPLAY_FILLED_RECTANGLE});
+                {r2d2::frame_type::DISPLAY_RECTANGLE,
```

Comment:
Sidenote; are you planning on adding more frames types? Internal communication currently has a limit of 8 I believe; let Koen or me know if you're going to run into this limit.

Reply:
at this point in time the display is finished with all the packets that are already existing we will let you or koen know if we need more